lib/matplotlib/tests/test_axes.py

```diff
@@ -9452,3 +9452,35 @@ def test_wrong_use_colorizer():
     for kwrd in kwrds:
         with pytest.raises(ValueError, match=match_str):
             fig.figimage(c, colorizer=cl, **kwrd)
+
+
+def test_bar_color_precedence():
+    # Test the precedence of 'color' and 'facecolor' in bar plots
+    to_rgba = mcolors.to_rgba
+    fig, ax = plt.subplots()
+
+    # Case 1: Only 'color'
+    bars = ax.bar([1, 2, 3], [4, 5, 6], color='red')
+    for bar in bars:
+        assert to_rgba(bar.get_facecolor()) == to_rgba('red')
+
+    ax.cla()
```

Comment:
You don't need to clear in between, which unfortunately is a relatively expensive action. Instead I'd do

```
    # case 1: no color specified
    bars = ax.bar([1, 2, 3], [4, 5, 6])
    [...]

    # case 2: Only 'color'
    bars = ax.bar([11, 12, 13], [4, 5, 6], color='red')
    [...]

    # case 3: Only 'color'
    bars = ax.bar([21, 22, 23], [4, 5, 6], color='red')
    [...]

    # case 4: 'facecolor' and 'color'
    bars = ax.bar([1, 2, 3], [4, 5, 6], color='red', facecolor='green')
    [...]
```

Putting the no color case first ensures noting is messing with the colorcycle.

Optional: You could use https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.same_color.html#matplotlib.colors.same_color for the asserts.